test/negative/spirv_report_bad_input.spt

```diff
@@ -1,33 +0,0 @@
-; RUN: llvm-spirv %s -to-binary -o %t.spv
```

Comment:
Why is this test being removed?
Just for understanding. Not a blocking question.
Thanks

Reply:
Report API was added as completely separated parsing pipeline. This fact also means that it needed it's own checks against malformed SPIR-V binary file. The removed file tested these checks. In this PR I'm combining Reporting API with the main parser which also means that it's the subject of the same checks and guards as the hot path pipeline. Main parser is tested already in other tests and IMO doesn't need another one.